### PR TITLE
Draft: Split search bar from item specific behaviour

### PR DESCRIPTION
**Pull request by ryan-rushton**
POC to make way for the search bar with autocomplete being used for loadouts and mods.

### CODE REVIEW

File: src/app/search/ItemSearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    </Portal>
+  );
+}
+
+/**
+ * A reusable, autocompleting item search input. This is an uncontrolled input that
+ * announces its query has changed only after some delay. This is the new version of the component
+ * that offers a browser-style autocompleting search bar with history.
+ *
+ * TODO: Should this be the main search bar only, or should it also work for item picker, etc?
```
Comment (bhollis) [inline src/app/search/ItemSearchBar.tsx:102]: The intention is that it works anywhere we can search items

**Review by bhollis — COMMENTED** [review]
Seems OK but I'm not sure exactly where this is going. This component is really a monster and I'm not sure that breaking it up into these specific layers is exactly what'll help disconnect them. Maybe another way to say it is I'm not sure this parameterizes the basic idea of an autocompleting search bar enough to get it to be generic for loadouts, mods, etc. The only thing I've managed to figure out so far is that we'll want to generate different versions of the SearchConfig for those different searches, and pass that to the search bar.

File: src/app/search/ItemSearchBar.tsx (hunk 1 above)
Comment (ryan-rushton) [reply src/app/search/ItemSearchBar.tsx:102]: That comment was copy paste from the old version I am pretty sure

**Comment by ryan-rushton** [comment]
K, I might just straight up build a loadouts search bar using the current one as a template instead then. Will be easier to see what needs extracting after we have a second implementation.

**Pull request closed without merging**